src/chat/chat_panel.rs

```diff
@@ -394,15 +394,25 @@ pub struct ChatPanel {
 
     #[rust(false)]
     focus_on_prompt_input_pending: bool,
+
+    #[rust]
+    current_chat_id: Option<ChatID>,
 }
 
 impl Widget for ChatPanel {
     fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
+        let store = scope.data.get_mut::<Store>().unwrap();
+
+        if store.chats.get_current_chat_id() != self.current_chat_id {
+            self.current_chat_id = store.chats.get_current_chat_id();
+            self.reset_scroll_messages(store);
+            self.redraw(cx);
+        }
+
         self.view.handle_event(cx, event, scope);
         self.widget_match_event(cx, event, scope);
 
-        let store = scope.data.get_mut::<Store>().unwrap();
-        self.update_state(store);
+        self.update_state(scope);
```

Comment:
Mainly changed to pass scope and not store down, or it would need to get the store 2 times. You can't just use one cause the store would be needed at the start and at the end, but the middle instructions requires scope, which would cause borrowing issues.